docs/source/site-specific-config.rst

```diff
@@ -0,0 +1,142 @@
+.. _site_specific_config:
+
+Site-Specific Configuration
+***************************
+A site might have compilers that Fab doesn't know about, or prefers
+a different compiler from the Fab default. Fab abstracts the compilers
+and other programs required during building as an instance of a
+:class:`~fab.tools.tool.Tool` class. All tools that Fab knows about, are
+available in a :class:`~fab.tools.tool_repository.ToolRepository`.
+That will include tools that might not be available on the current system.
+
+Each tool belongs to a certain category of
+:class:`~fab.tool.categories.Categories`. A `ToolRepository` can store
+several instances of the same category.
+
+At build time, the user has to create an instance of
+:class:`~fab.tools.tool_box.ToolBox` and pass
+it to the :class:`~fab.build_config.BuildConfig` object. This toolbox
+contains all the tools that will be used during the build process, but
+it can only store one tool per category. If a certain tool should not
+be defined in the toolbox, the default from the `ToolRepository` will
+be used. This is useful for many standard tools like `git`, `rsync`
+etc that de-facto will never be changed.
+
+
+Categories
+==========
+All possible categories are defined in
+:class:`~fab.tool.categories.Categories`. If additional categories
+should be required, they can be added.
```

Comment:
That needs to modify the sources, but see #311, where I suggest to generalise the usage of `MISC`, meaning user can easily add tools to the `MISC` category.

Alternative, we could allow `Union[str, Categories]`, then a user can add arbitrary strings, and it would certainly not interfere with the Fab-hard-coded categories.